src/index.js

```diff
@@ -0,0 +1,195 @@
+        skySelect.value = "sunny";
+        updateSkyEmoji("sunny");
+        sky.textContent = "☁️ ☁️ ☁️ ☀️ ☁️ ☁️";
+
+    } else if (tempF >= 70) {
+        tempValue.classList.add("orange");
+        landscape.textContent = "🌿🌻🌼🌸🌷🪻☀️🦋🐝🌤️";
+        skySelect.value = "sunny";
+        updateSkyEmoji("sunny");
+        sky.textContent = "☁️ ☁️ ☁️ ☀️ ☁️ ☁️";
+
+    } else if (tempF >= 60) {
+        tempValue.classList.add("yellow");
+        landscape.textContent = "🌳🍂🪵🍃🦉🍁🐿️🍄🍁⛅";
+        skySelect.value = "cloudy";
+        updateSkyEmoji("cloudy");
+        sky.textContent = "☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️";
+
+    } else if (tempF >= 50) {
+        tempValue.classList.add("green");
+        landscape.textContent = "🍁🌦️🌾🌤️🐛🌼🌦️🍂🪲☘️";
+        skySelect.value = "cloudy";
+        updateSkyEmoji("cloudy");
+        sky.textContent = "☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️";
+
+    } else {
+        tempValue.classList.add("teal");
+        landscape.textContent = "❄️⛄🌨️🌬️⛄🌲☃️❄️🥶🌨️";
+        skySelect.value = "snowy";
+        updateSkyEmoji("snowy");
+        sky.textContent = "🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨";
+    }
+}
+
+increaseButton.addEventListener("click", function () {
+    tempF++;
+    updateTempColor();
+});
+
+decreaseButton.addEventListener("click", function () {
+    tempF--;
+    updateTempColor();
+});
+updateTempColor();
+
+
+function displayCityName() {
+    cityNameInput.addEventListener("input", () => {
+        headerCityName.textContent = cityNameInput.value;
+    });
+}
+displayCityName();
+
+// retry helper function
+// wave 4: //
+const getWithRetry = (url, queryParams, attempt = 1) => {
+    return axios.get(url, {
+        params: queryParams
+    }).then((response) => {
+        const data = response.data;
+
+        if (data.error) {
+            if (attempt >= 5) {
+                console.log("Max attempts reached!");
+                return null;
+            }
+            console.log(`Empty response on attempt ${attempt}. Retrying...`);
+            return new Promise((resolve) => {
+                setTimeout(() => {
+                    resolve(getWithRetry(url, queryParams, attempt + 1));
+                }, 600 * attempt);
+            });
+        }
+        return response;
+    })
+};
+
+const findLatitudeAndLongitude = (query) => {
+    const url = "http://localhost:5000/location";
+    return getWithRetry(url, { q: query, format: 'json' }).then((response) => {
+        if (!response.data) {
+            return null;
+        }
+        const { lat: latitude, lon: longitude } = response.data[0];
+        return { latitude, longitude };
+    });
+};
+const findWeatherLatLon = (lat, lon) => {
+    const url = "http://localhost:5000/weather";
+
+    return getWithRetry(url, { lat, lon }).then((response) => {
+        if (!response.data) {
+            return null;
+        }
+        const tempK = response.data.main.temp;
+        const tempC = Math.round(tempK - 273.15);
+        const tempF = Math.round((tempC * 9) / 5 + 32);
```

Comment:
This logic feels a little misplaced in this function since its main responsibility is to make the `axios` call to the weather endpoint. I'd say to more closely follow SRP, to move this logic else where (perfects inside of the event handler for the button?).